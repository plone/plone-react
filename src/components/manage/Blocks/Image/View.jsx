/**
 * View image block.
 * @module components/manage/Blocks/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import Image from '@plone/volto/components/theme/Image/Image';

import { flattenToAppURL, isInternalURL } from '@plone/volto/helpers';

/**
 * View image block class.
 * @class View
 * @extends Component
 */
const View = ({ data, detached }) => (
  <p
    className={cx(
      'block image align',
      {
        center: !Boolean(data.align),
        detached,
      },
      data.align,
    )}
  >
    {data.url && (
      <>
        {(() => {
          const image = (
            <Image
              image={data.url}
              alt={data.alt}
              className={cx({
                'full-width': data.align === 'full',
                large: data.size === 'l',
                medium: data.size === 'm',
                small: data.size === 's',
              })}
            />
          );
          if (data.href) {
            if (!isInternalURL(data.href)) {
              return (
                <a
                  target={data.openLinkInNewTab ? '_blank' : null}
                  href={data.href}
                >
                  {image}
                </a>
              );
            } else {
              return (
                <Link
                  to={flattenToAppURL(data.href)}
                  target={data.openLinkInNewTab ? '_blank' : null}
                >
                  {image}
                </Link>
              );
            }
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
