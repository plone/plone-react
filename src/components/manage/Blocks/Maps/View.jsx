/**
 * View map block.
 * @module components/manage/Blocks/Maps/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

/**
 * View image block class.
 * @class View
 * @extends Component
 */

const View = ({ data }) => (
  <p
    className={cx(
      'block maps align',
      {
        center: !Boolean(data.align),
      },
      data.align,
    )}
  >
    <div
      className={cx('video-inner', {
        'full-width': data.align === 'full',
      })}
    >
      <iframe
        title="Embeded Google Maps"
        src={data.url}
        className="google-map"
        frameBorder="0"
        allowFullScreen
      />
    </div>
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
