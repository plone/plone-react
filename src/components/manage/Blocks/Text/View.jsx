/**
 * View text block.
 * @module components/manage/Blocks/Text/View
 */

import PropTypes from 'prop-types';
import redraft from 'redraft';

import { settings } from '~/config';

/**
 * View text block class.
 * @class View
 * @extends Component
 */
const View = ({ data }) =>
  data.text
    ? redraft(data.text, settings.ToHTMLRenderers, settings.ToHTMLOptions)
    : '';

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
