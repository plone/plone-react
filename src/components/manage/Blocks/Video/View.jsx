/**
 * View video block.
 * @module components/manage/Blocks/Video/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Embed } from 'semantic-ui-react';
import cx from 'classnames';

/**
 * View video block class.
 * @class View
 * @extends Component
 */
const View = ({ data }) => (
  <p
    className={cx(
      'block video align',
      {
        center: !Boolean(data.align),
      },
      data.align,
    )}
  >
    {data.url && (
      <div
        className={cx('video-inner', {
          'full-width': data.align === 'full',
        })}
      >
        {data.url.match('list') ? (
          <Embed
            url={`https://www.youtube.com/embed/videoseries?list=${
              data.url.match(/^.*\?list=(.*)$/)[1]
            }`}
            icon="arrow right"
            defaultActive
            autoplay={false}
          />
        ) : (
          <Embed
            id={
              data.url.match(/.be\//)
                ? data.url.match(/^.*\.be\/(.*)/)[1]
                : data.url.match(/^.*\?v=(.*)$/)[1]
            }
            source="youtube"
            icon="arrow right"
            defaultActive
            autoplay={false}
          />
        )}
      </div>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
