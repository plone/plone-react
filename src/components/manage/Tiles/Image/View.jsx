/**
 * View image tile.
 * @module components/manage/Tiles/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import { settings } from '~/config';

import { flattenToAppURL } from '@plone/volto/helpers';

/**
 * View image tile class.
 * @class View
 * @extends Component
 */
const View = ({ data, detached }) => (
  <p
    className={cx(
      'tile image align',
      {
        center: !Boolean(data.align),
        detached,
      },
      data.align,
    )}
  >
    {data.url && (
      <>
        {(() => {
          const image = (
            <img
              src={
                data.url.includes(settings.apiPath)
                  ? `${flattenToAppURL(data.url)}/@@images/image`
                  : data.url
              }
              alt={data.alt || ''}
            />
          );
          if (data.external) {
            const isReallyExternal =
              (data.external.startsWith('http') ||
                data.external.startsWith('https')) &&
              !data.external.includes(settings.apiPath);

            if (isReallyExternal) {
              return (
                <a
                  target={data.openLinkInNewTab ? '_blank' : null}
                  href={data.external}
                >
                  {image}
                </a>
              );
            } else {
              return (
                <Link
                  to={data.external.replace(settings.apiPath, '')}
                  target={data.openLinkInNewTab ? '_blank' : null}
                >
                  {image}
                </Link>
              );
            }
          } else if (data.href) {
            return (
              <Link
                to={data.href}
                target={data.openLinkInNewTab ? '_blank' : null}
              >
                {image}
              </Link>
            );
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
