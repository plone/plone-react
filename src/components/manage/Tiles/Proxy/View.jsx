/**
 * View Proxy tile.
 * @module components/manage/Tiles/Proxy/View
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { get } from 'lodash';
import { Image, Item } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

import { settings } from '~/config';

import { getContent, resetContent } from '../../../../actions';

/**
 * Proxy view component.
 * @class View
 * @extends Component
 */
export class View extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    tile: PropTypes.string.isRequired,
    data: PropTypes.objectOf(PropTypes.any).isRequired,
    getContent: PropTypes.func.isRequired,
    resetContent: PropTypes.func.isRequired,
    contentSubrequests: PropTypes.objectOf(PropTypes.any),
  };

  static defaultProps = {
    contentSubrequests: {},
  };

  /**
   * Component did mount
   * @method componentDidMount
   * @returns {undefined}
   */
  componentDidMount() {
    const { selectedItem } = this.props.data;
    if (selectedItem) {
      // Use subrequests to fetch tile data
      this.props.getContent(selectedItem, undefined, this.props.tile);
    }
  }

  /**
   * Component will unmount. Reset loaded content.
   * @method componentWillUnmount
   * @returns {undefined}
   */
  componentWillUnmount() {
    this.props.resetContent(this.props.tile);
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    const { contentSubrequests, tile } = this.props;
    const selectedItem = get(this.props, 'data.selectedItem', '');

    // Using null as default for consistency with content reducer
    // see reducers/content/content.js, look for action GET_CONTENT_PENDING
    const data = get(contentSubrequests, [tile, 'data'], null);
    if (data && Object.keys(data).length) {
      const image = get(data, 'image.scales.mini.download', '').replace(
        settings.apiPath,
        '',
      );

      return (
        <Item.Group className="proxy-item">
          <Item>
            {image && (
              <Link to={selectedItem} className="ui small image">
                <Image src={image} alt={data.title} />
              </Link>
            )}
            <Item.Content>
              <Item.Header>
                <Link to={selectedItem}>{data.title}</Link>
              </Item.Header>
              <Item.Description>{data.description}</Item.Description>
            </Item.Content>
          </Item>
        </Item.Group>
      );
    }
    return null;
  }
}

export default connect(
  state => ({
    contentSubrequests: state.content.subrequests,
  }),
  dispatch =>
    bindActionCreators(
      {
        getContent,
        resetContent,
      },
      dispatch,
    ),
)(View);
