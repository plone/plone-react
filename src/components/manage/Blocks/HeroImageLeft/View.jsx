/**
 * View image block.
 * @module components/manage/Blocks/Hero/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import Image from '@plone/volto/components/theme/Image/Image';

/**
 * View image block class.
 * @class View
 * @extends Component
 */
const View = ({ data }) => (
  <div className="block hero">
    <div className="block-inner-wrapper">
      {data.url && <Image image={data.url} className="hero-image" />}
      <div className="hero-body">
        {data.title && <h1>{data.title}</h1>}
        {data.description && <p>{data.description}</p>}
      </div>
    </div>
  </div>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
