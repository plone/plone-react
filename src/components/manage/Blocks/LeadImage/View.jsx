/**
 * View image block.
 * @module components/manage/Blocks/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import { flattenToAppURL } from '@plone/volto/helpers';
import Image from '@plone/volto/components/theme/Image/Image';

import config from '@plone/volto/registry';

/**
 * View image block class.
 * @class View
 * @extends Component
 */
const View = ({ data, properties }) => (
  <p
    className={cx(
      'block image align',
      {
        center: !Boolean(data.align),
      },
      data.align,
    )}
  >
    {properties.image && (
      <>
        {(() => {
          const image = (
            <Image
              image={properties.image}
              alt={properties.image_caption}
              className={cx({ 'full-width': data.align === 'full' })}
            />
          );
          if (data.href) {
            if (
              (data.href.startsWith('http') || data.href.startsWith('https')) &&
              !data.href.includes(config.settings.apiPath)
            ) {
              return (
                <a
                  target={data.openLinkInNewTab ? '_blank' : null}
                  href={data.href}
                >
                  {image}
                </a>
              );
            } else {
              return (
                <Link
                  to={flattenToAppURL(data.href)}
                  target={data.openLinkInNewTab ? '_blank' : null}
                >
                  {image}
                </Link>
              );
            }
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  properties: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
