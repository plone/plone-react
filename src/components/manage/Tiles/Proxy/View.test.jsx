import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import { settings } from '~/config';

import { View } from './View';

const props = {
  data: {
    selectedItem: '/test',
  },
  getContent: () => {},
  resetContent: () => {},
  tile: 'test',
  contentSubrequests: {
    test: {
      data: {
        '@id': `${settings.apiUrl}/test`,
        image: {
          scales: {
            mini: {
              download: `${settings.apiUrl}/test/image.jpeg`,
            },
          },
        },
        title: 'Test',
        description: 'Proxy',
      },
    },
  },
};

test('renders a proxy view component', () => {
  const component = renderer.create(
    <MemoryRouter>
      <View {...props} />
    </MemoryRouter>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
