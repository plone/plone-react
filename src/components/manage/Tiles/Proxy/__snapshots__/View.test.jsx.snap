// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a proxy view component 1`] = `
<div
  className="ui items proxy-item"
>
  <div
    className="item"
  >
    <a
      className="ui small image"
      href="/test"
      onClick={[Function]}
    >
      <img
        alt="Test"
        className="ui image"
        src="undefined/test/image.jpeg"
      />
    </a>
    <div
      className="content"
    >
      <div
        className="header"
      >
        <a
          href="/test"
          onClick={[Function]}
        >
          Test
        </a>
      </div>
      <div
        className="description"
      >
        Proxy
      </div>
    </div>
  </div>
</div>
`;
